src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -1754,58 +1791,271 @@ public override bool MightContainExtensionMethods
             }
         }
 
+#nullable enable
         public override TypeKind TypeKind
         {
             get
             {
                 TypeKind result = _lazyKind;
-
                 if (result == TypeKind.Unknown)
```

Comment:
>if (result == TypeKind.Unknown) [](http://example.com/codeflow?start=16&length=31)

Consider moving the logic around ```_lazyKind``` to ```MakeTypeKindAndBaseTypeFromMetadata``` in order to avoid recalculating the kind when ```MakeTypeKindAndBaseTypeFromMetadata``` is called by ```GetDeclaredBaseType``` and ```_lazyKind``` already has value.

Reply:
Reverted changes and removed `maybeExtension()` following offline discussion.